src/renderer/screens/FirmwareUpdate.js

```diff
@@ -58,17 +59,25 @@ const FirmwareUpdate = (props) => {
 
   const focusDeviceDescriptor =
     props.focusDeviceDescriptor || focus.focusDeviceDescriptor;
-  const isBootloader = focusDeviceDescriptor.bootloader;
 
+  const [isBootloader, setIsBootloader] = useState(false);
   const [factoryConfirmationOpen, setFactoryConfirmationOpen] = useState(false);
   const [confirmationOpen, setConfirmationOpen] = useState(false);
   const [activeStep, setActiveStep] = useState(-1);
   const [flashSteps, setFlashSteps] = useState([]);
   const [progress, setProgress] = useState("idle");
-  const [factoryReset, setFactoryReset] = useState(isBootloader);
+  const [factoryReset, setFactoryReset] = useState(false);
 
   const { t } = useTranslation();
 
+  useEffectOnce(() => {
+    const initialize = async () => {
+      setIsBootloader(focusDeviceDescriptor.bootloader);
+      setFactoryReset(focusDeviceDescriptor.bootloader);
```

Comment:
This wants a comment explaining the intent. "If we're starting off in bootloader mode, default to a factory reset because _____"

After that, merge at wlll.